[Response Ops] [Rule Form] Add new flyout to rule form library, responsive design and illustration to rule form page

Zacqary opened the pull request:
## This PR is DEPENDENT on https://github.com/elastic/kibana/pull/205944

Because we use a fork-and-pull contribution structure at Elastic, Github is not able to base this PR off the [correct branch](https://github.com/Zacqary/kibana/tree/195211-form-steps-hook) without opening the PR outside the main Kibana repo. Please use manual diffing tools to review this code until #205944 merges

## Summary

Part of #195211

### Non-User Facing Changes
- Adds the new rule flyout to `@kbn/response-ops-rule-form`. ***It is not yet actually user-facing anywhere in the application, this will be done in a second PR.***
<details>
<summary><h4>Screenshots</h4></summary>
<img width="508" alt="Screenshot 2025-01-08 at 4 29 55 PM" src="https://github.com/user-attachments/assets/7f03cd3a-5f37-4ac2-9992-ca4951664770" />
<img width="502" alt="Screenshot 2025-01-08 at 4 29 59 PM" src="https://github.com/user-attachments/assets/c0620fc2-83db-4603-90b7-1282e5b0e6ab" />
<img width="507" alt="Screenshot 2025-01-08 at 4 30 03 PM" src="https://github.com/user-attachments/assets/8440d551-46af-49e0-9c92-22d6b3ba1866" />
<img width="507" alt="Screenshot 2025-01-08 at 4 30 32 PM" src="https://github.com/user-attachments/assets/cf7493a7-6e4a-4e55-8027-89e9b36012fc" />

</details>


### User-Facing Changes
These changes were added to the existing full page rule form to minimize the amount of code differences between the flyout and the full page

- Adds some responsive styling to the rule form page to make it look more similar to the flyout when the browser window is narrow
<details>
<summary>Screenshot</summary>
<img width="783" alt="Screenshot 2025-01-08 at 4 31 50 PM" src="https://github.com/user-attachments/assets/a3532b92-9f22-4e88-bcc3-e408fc53e64c" />

</details>

- Adds the new illustrated "Add an action" empty prompt from the flyout designs to the existing rule form page
<details>
<summary>Screenshot</summary>
<img width="1299" alt="Screenshot 2025-01-08 at 5 00 55 PM" src="https://github.com/user-attachments/assets/c4acd50d-9268-4874-b650-ecba532f3e9c" />

</details>

### Testing

To test the new flyout, edit `packages/response-ops/rule_form/src/create_rule_form.tsx` and `packages/response-ops/rule_form/src/edit_rule_form.tsx` so that they render `<RuleFlyout>` instead of `<RulePage>`. 

<details>
<summary><strong>Use this diff block</strong></summary>

```diff
diff --git a/packages/response-ops/rule_form/src/create_rule_form.tsx b/packages/response-ops/rule_form/src/create_rule_form.tsx
index 2f5e0472dcd..564744b96ec 100644
--- a/packages/response-ops/rule_form/src/create_rule_form.tsx
+++ b/packages/response-ops/rule_form/src/create_rule_form.tsx
@@ -31,6 +31,7 @@ import {
   parseRuleCircuitBreakerErrorMessage,
 } from './utils';
 import { RULE_CREATE_SUCCESS_TEXT, RULE_CREATE_ERROR_TEXT } from './translations';
+import { RuleFlyout } from './rule_flyout';
 
 export interface CreateRuleFormProps {
   ruleTypeId: string;
@@ -199,7 +200,7 @@ export const CreateRuleForm = (props: CreateRuleFormProps) => {
           }),
         }}
       >
-        <RulePage isEdit={false} isSaving={isSaving} onCancel={onCancel} onSave={onSave} />
+        <RuleFlyout isEdit={false} isSaving={isSaving} onCancel={onCancel} onSave={onSave} />
       </RuleFormStateProvider>
     </div>
   );
diff --git a/packages/response-ops/rule_form/src/edit_rule_form.tsx b/packages/response-ops/rule_form/src/edit_rule_form.tsx
index 392447114ed..41aecd7245a 100644
--- a/packages/response-ops/rule_form/src/edit_rule_form.tsx
+++ b/packages/response-ops/rule_form/src/edit_rule_form.tsx
@@ -26,6 +26,7 @@ import {
 import { RULE_EDIT_ERROR_TEXT, RULE_EDIT_SUCCESS_TEXT } from './translations';
 import { getAvailableRuleTypes, parseRuleCircuitBreakerErrorMessage } from './utils';
 import { DEFAULT_VALID_CONSUMERS, getDefaultFormData } from './constants';
+import { RuleFlyout } from './rule_flyout';
 
 export interface EditRuleFormProps {
   id: string;
@@ -193,7 +194,7 @@ export const EditRuleForm = (props: EditRuleFormProps) => {
           showMustacheAutocompleteSwitch,
         }}
       >
-        <RulePage isEdit={true} isSaving={isSaving} onSave={onSave} onCancel={onCancel} />
+        <RuleFlyout isEdit={true} isSaving={isSaving} onSave={onSave} onCancel={onCancel} />
       </RuleFormStateProvider>
     </div>
   );
```

</details>

### Still Todo

1. Replace the action connector modal with an in-flyout UI as called for in the [design spec](https://www.figma.com/design/zetHXnUP0YnDG4YmvPwRb8/Adapt-new-Rule-form-to-work-in-flyout)
2. Add the Show Request UI
3. Replace all instances of the v1 rule flyout with this new one (it's used heavily in solutions, not in Stack Management)

### Checklist
- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

Zacqary committed:
Move rule form steps to hook with progress tracking

Zacqary committed:
Add type annotations to rule form step hooks

Zacqary committed:
Add useRuleFormSteps test

Zacqary committed:
Add new rule flyout to response-ops pkg

Zacqary committed:
Fix rule page mocks

Zacqary committed:
Add responsive rule form layout with container query

Zacqary committed:
Add empty prompt with illustration for stration when no actions

elasticmachine committed:
Merge branch 'main' into 195211-form-steps-hook

Zacqary committed:
Restore original create page

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265425)
* Commit: 18462f5253a4856894e33e223713c6536d405c61

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/265425#0194482a-cbe8-4496-af44-ba100efb12ff)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265425","number":265425,"commit":"18462f5253a4856894e33e223713c6536d405c61"}],"number":265425}
buildkite-pr-comment-->

mbondyra committed:
[Dashboard][Collapsable Panels] Respond to touch events (#204225)

## Summary

Adds support to touch events. The difference between these ones and
mouse events is that once they are active, the scroll is off (just like
in the current Dashboard)


https://github.com/user-attachments/assets/4cdcc850-7391-441e-ab9a-0abbe70e4e56

Fixes https://github.com/elastic/kibana/issues/202014

cqliu1 committed:
[Inspector] EUI Visual Refresh Integration (#204436)

## Summary

Related to https://github.com/elastic/kibana/issues/203132.
Closes [#204595](https://github.com/elastic/kibana/issues/204595).

This replaces all references to euiThemeVars in favor of the useEuiTheme
hook in the inspector plugin.


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [ ] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard
to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified
risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

stephmilovic committed:
[Security solution] CSS emotion/react fix (#205934)

kertal committed:
[Discover] Unskip functional discover request counts test and skip ES|QL part (#205690)

Unskipping the main test, and unskipping the `ES|QL` part due to flakiness

pgayvallet committed:
[inference] handle toolCall indices not starting at 0 (#205954)

## Summary

Turns out, claude can in some situations (when returning both text and
toolcall in a single message) starts their toolcall index at `1` instead
of `0`, which introducing null values in the concatenated messages.

This fixes it, by removing null values from the tool calls when merging
the chunks.

Also remove the SKA codeowner override for the inference plugin to get
back the shared ownership

pgayvallet committed:
[NL-to-ESQL] update internal documentation (#205853)

## Summary

Fix https://github.com/elastic/kibana/issues/205606

- Re-generate the internal ES|QL documentation using the generation
script (+ human review)
- Add more scenario to the NL-to-ESQL evaluation suite  
- Some prompt engineering
  - improving the system instructions / functions summary
  - add more examples to the summary
  - adapt a few opinionated examples for some specific functions 

## Evaluation

- average based on 4 runs for each model/branch tuple
- the new tests were locally added to main to run against the same suite
and properly evaluate the difference

| Model  | before (main) | after (PR) | delta |
| ------------- | ------------- | ------------- | ------------- |
| GPT-4o  | 90.9 | 97.74  | + 6.84 |
| Claude 3.5 Sonnet v2  |  88.58 | 96.49 | +7.91 |
| Gemini 1.5-pro-002  | 88.17  | 94.19 | +6.02 |

Overall, the prompt engineering somewhat significantly improved the
generation efficiency.

MiriamAparicio committed:
[ObsUX] Use styled from @emotion in log-stream component (#205918)

Closes https://github.com/elastic/kibana/issues/205683

LogStream embeddable loads correctly

<img width="1486" alt="Screenshot 2025-01-08 at 15 08 46"
src="https://github.com/user-attachments/assets/7e53e7ba-2b86-471d-9fb0-39ae54b7f919"
/>

flash1293 committed:
🌊 Streams: Show data retention on stream (#204125)

Show data retention on streams

In case of a policy, the name of the policy is shown (badge is clickable
and leads to the edit page of the policy):
<img width="524" alt="Screenshot 2024-12-12 at 20 57 36"
src="https://github.com/user-attachments/assets/2664b45b-2473-49c4-b1d6-dccb8fe48d43"
/>

In case of DLM, the effect retention is shown:
<img width="532" alt="Screenshot 2024-12-12 at 20 58 42"
src="https://github.com/user-attachments/assets/07ca8086-75e2-45f8-9d71-17bd0a76ebe5"
/>

This is just the display piece, editing retention will be added later
on.

This PR adjusts the base streams data stream settings to use a localized
data stream retention configuration to make it compatible with
serverless.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

haetamoudi committed:
[Automatic import] readme input types templates (#194308)

viajes7 committed:
[Fleet] Fix required validation for multi text input field (#205768)

## Summary

Fixes #196648 

- ✅  add required validation to blank spaces value
- ❓ when adding additional rows for path. It seems that the updated
value is specially processed.

**And this is why the required validation is not triggered when clicking
to add a row.**


![image](https://github.com/user-attachments/assets/f2f0d813-3762-47c5-bf31-1171c82c3b07)

## After fixup


![image](https://github.com/user-attachments/assets/857bef47-38c1-4951-8ea9-15b5ef5d143d)

ElenaStoeva committed:
[Data Streams] Improve jest tests for bulk actions (#204938)

Follow-up to https://github.com/elastic/kibana/pull/203083

## Summary

This PR improves the jest tests for bulk actions in data streams.

afharo committed:
[Upgrade Assistant] Reindexing optimizations (#205055)

miloszmarcinkowski committed:
[Profiling UI] Remove redundant page separator (#206004)

Closes #206002

## Summary

This PR removes redundant page separator to save up some space.

Before:

<img width="1727" alt="Screenshot 2025-01-08 at 16 16 50"
src="https://github.com/user-attachments/assets/9f69ebb2-d1a7-430e-9d9e-376712909ac1"
/>

After: 

<img width="1728" alt="Screenshot 2025-01-08 at 16 19 34"
src="https://github.com/user-attachments/assets/9d738aa0-adba-46f3-ba79-5f20b7b90499"
/>

adcoelho committed:
[ResponseOps][Rules] Moved params of anomaly detection rules to a separate package (#205497)

Connected with #195189

## Summary

- Moved params of anomaly detection rule type to
`/response-ops/rule_params/anomaly_detection/`
- Moved params of anomaly detection jobs health rule type to
`/response-ops/rule_params/anomaly_detection_jobs_health/`

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

angorayc committed:
[Security Solution][threat hunting explore] EUI refresh: Remove custom color hex (#204631)

## Summary

https://github.com/elastic/kibana/issues/202498
https://github.com/elastic/kibana/issues/202503

1. This PR does **Not** include the severity color change. It will be
implemented in a follow up PR once color tokens have been decided.
2. This PR updates the type `GetLensAttributes` to accept `euiTheme`, so
all the functions in this type are updated accordingly.

https://github.com/elastic/kibana/pull/204631/files#diff-abe20658865cad59eadcff945552b40832d96da0264ed89ddd5ab25ded1420a3R30

----

## To test:

Please verify if visualizations are displayed properly.

### Running Kibana with the Borealis theme
In order to run Kibana with `Borealis`, you'll need to do the following:

1. Set the following in kibana.dev.yml:
`uiSettings.experimental.themeSwitcherEnabled: true`
4. Run Kibana with the following environment variable set: 
```KBN_OPTIMIZER_THEMES="borealislight,borealisdark,v8light,v8dark" yarn start```

5. This will expose a toggle under Stack Management > Advanced Settings > Theme version, which you can use to toggle between Amsterdam and Borealis.


![Image](https://github.com/user-attachments/assets/78d64946-43fc-4400-bbb1-229d900b7f05)

----


### Explore


<img width="2557" alt="host_after" src="https://github.com/user-attachments/assets/f69b6e2a-58f6-4ed4-9f38-dcdbcf9919ed" />
<img width="1281" alt="users_after" src="https://github.com/user-attachments/assets/0eec8e9c-e678-4a66-83ee-4b8d11380b8e" />
<img width="2552" alt="network_dns" src="https://github.com/user-attachments/assets/19f06d2a-6e51-419b-9f89-233bfa5727ba" />
<img width="2557" alt="network_after" src="https://github.com/user-attachments/assets/3b90c5e2-23a1-4f15-a2d0-f9290a39af30" />

----

### Dashboards

<img width="2557" alt="dashboard_overview" src="https://github.com/user-attachments/assets/c1873359-fee9-42c6-addd-fe2bc1c98aee" />
<img width="2558" alt="dashboard_detection_response_2" src="https://github.com/user-attachments/assets/f6853710-0938-402b-b326-fa00d586b5d6" />
<img width="2559" alt="dashboard_detection_response_1" src="https://github.com/user-attachments/assets/4eb75526-9a57-46e9-b090-b53353956ea1" />

----


### Alerts
<img width="2555" alt="alerts_chart_collapsed" src="https://github.com/user-attachments/assets/6ecf5dd5-a785-4701-900b-0454f024b36d" />
<img width="2554" alt="summary" src="https://github.com/user-attachments/assets/1731a6ea-ef2b-4d7d-bf21-4041e59f0ad4" />
<img width="2559" alt="trend" src="https://github.com/user-attachments/assets/b9a741d1-a359-4273-9555-850cdcbc8932" />
<img width="2557" alt="counts" src="https://github.com/user-attachments/assets/a6193ccc-86b8-4974-ad9f-9417e200e859" />
<img width="1281" alt="treemap" src="https://github.com/user-attachments/assets/7b6e163a-a660-4bb1-a6de-88e21934b98a" />

----

### Rules preview
<img width="2556" alt="Screenshot 2024-12-18 at 13 45 33" src="https://github.com/user-attachments/assets/47099c18-86ee-455a-a5af-ebd6a29904a5" />


----

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

angorayc committed:
[Security Solution][threat hunting explore] EUI refresh: Rename color variables (#204908)

## Summary

https://github.com/elastic/kibana/issues/202499

### Running Kibana with the Borealis theme
In order to run Kibana with Borealis, you'll need to do the following:

Set the following in kibana.dev.yml:
uiSettings.experimental.themeSwitcherEnabled: true

Run Kibana with the following environment variable set:
KBN_OPTIMIZER_THEMES="borealislight,borealisdark,v8light,v8dark" yarn
start

This will expose a toggle under Stack Management > Advanced Settings >
Theme version, which you can use to toggle between Amsterdam and
Borealis.


![Image](https://github.com/user-attachments/assets/78d64946-43fc-4400-bbb1-229d900b7f05)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

albertoblaz committed:
Turn AssetInventory sub-plugin into a Security solution page (#205363)

## Summary

Turn AssetInventory plugin into a simple Security solution page based on
this conversation:
- https://github.com/elastic/security-team/issues/10346

Follow-up of this work, where we initially developed the Asset Inventory
bare-bones as a plugin:
- https://github.com/elastic/kibana/issues/201704

Part of this work stream though not originally planned:
- https://github.com/elastic/security-team/issues/11247?reload=1

### Motivation

Reasoning behind is enabling fast development changes without any major
refactors in other packages/components that need to be reused, which
seems impossible as of now with the current codebase.

### Checklist

- [x] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking
changes have been approved by the breaking-change committee. The
`release_note:breaking` label should be applied in these situations.
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

In the future, we'll probably have to turn AssetInventory into a
sub-plugin again. Doing so doesn't bring any risk in the present, and
should only involve the re-generation of the deleted files in this PR.

---------

Co-authored-by: Paulo Henrique <paulo.henrique@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elastic-renovate-prod[bot] committed:
Update dependency @xyflow/react to ^12.3.6 (main) (#204459)

This PR contains the following updates:

| Package | Type | Update | Change |
|---|---|---|---|
| [@xyflow/react](https://reactflow.dev)
([source](https://togithub.com/xyflow/xyflow/tree/HEAD/packages/react))
| dependencies | patch | [`^12.3.5` ->
`^12.3.6`](https://renovatebot.com/diffs/npm/@xyflow%2freact/12.3.5/12.3.6)
|

---

### Release Notes

<details>
<summary>xyflow/xyflow (@&#8203;xyflow/react)</summary>

###
[`v12.3.6`](https://togithub.com/xyflow/xyflow/blob/HEAD/packages/react/CHANGELOG.md#1236)

[Compare
Source](https://togithub.com/xyflow/xyflow/compare/@xyflow/react@12.3.5...@xyflow/react@12.3.6)

##### Patch Changes

- [#&#8203;4846](https://togithub.com/xyflow/xyflow/pull/4846)
[`75017939`](https://togithub.com/xyflow/xyflow/commit/7501793900fdb125fe906fa3d9d83f1bd11c6e20)
Thanks [@&#8203;moklick](https://togithub.com/moklick)! - Make it
possible to use expandParent with immer and other immutable helpers

- [#&#8203;4865](https://togithub.com/xyflow/xyflow/pull/4865)
[`2c4acc2b`](https://togithub.com/xyflow/xyflow/commit/2c4acc2bd9ec271468bd4c904e19d7fca627d9e1)
Thanks [@&#8203;moklick](https://togithub.com/moklick)! - Add group node
to BuiltInNode type. Thanks
[@&#8203;sjdemartini](https://togithub.com/sjdemartini)!

- [#&#8203;4877](https://togithub.com/xyflow/xyflow/pull/4877)
[`9a8309da`](https://togithub.com/xyflow/xyflow/commit/9a8309dab892f047fce10d3c763466cf84f726b0)
Thanks [@&#8203;peterkogo](https://togithub.com/peterkogo)! - Fix
intersections for nodes with origins other than \[0,0]. Thanks
[@&#8203;gmvrpw](https://togithub.com/gmvrpw)!

- [#&#8203;4844](https://togithub.com/xyflow/xyflow/pull/4844)
[`6f11e552`](https://togithub.com/xyflow/xyflow/commit/6f11e552c3f51b5931b98d5ff798156cdb8cc2d9)
Thanks [@&#8203;moklick](https://togithub.com/moklick)! - Allow custom
data-testid for ReactFlow component

- [#&#8203;4816](https://togithub.com/xyflow/xyflow/pull/4816)
[`43aa52a8`](https://togithub.com/xyflow/xyflow/commit/43aa52a8cd93e6a01c01afa314c7036eb5b77131)
Thanks [@&#8203;moklick](https://togithub.com/moklick)! - Type
isValidConnection prop correctly by passing EdgeType

- [#&#8203;4855](https://togithub.com/xyflow/xyflow/pull/4855)
[`106c2cf8`](https://togithub.com/xyflow/xyflow/commit/106c2cf8e5e9a944ea7ea9a2558e26cb75f9ac93)
Thanks [@&#8203;mhuggins](https://togithub.com/mhuggins)! - Support
passing `path` element attributes to `BaseEdge` component.

- [#&#8203;4862](https://togithub.com/xyflow/xyflow/pull/4862)
[`adf4fb4e`](https://togithub.com/xyflow/xyflow/commit/adf4fb4e7bb136cce47ef540e9bcc0f82cf17ff4)
Thanks [@&#8203;bcakmakoglu](https://togithub.com/bcakmakoglu)! -
Prevent default scrolling behavior when nodes or a selection is moved
with an arrow key press.

- [#&#8203;4875](https://togithub.com/xyflow/xyflow/pull/4875)
[`41d4743a`](https://togithub.com/xyflow/xyflow/commit/41d4743a690f054f6dc2b9084b5fad8e54ade921)
Thanks [@&#8203;peterkogo](https://togithub.com/peterkogo)! - Prevent
unnecessary rerenders of edges when resizing the flow.

- [#&#8203;4826](https://togithub.com/xyflow/xyflow/pull/4826)
[`5f90acda`](https://togithub.com/xyflow/xyflow/commit/5f90acdab1f4748bb04f8b1ec67d569983684058)
Thanks [@&#8203;chrtze](https://togithub.com/chrtze)! - Forward ref of
the div inside Panel components.

- Updated dependencies
\[[`d60331e6`](https://togithub.com/xyflow/xyflow/commit/d60331e6baa7931c46af219e35c1bedbd156187c)]:
-
[@&#8203;xyflow/system](https://togithub.com/xyflow/system)[@&#8203;0](https://togithub.com/0).0.47

</details>

---

### Configuration

📅 **Schedule**: Branch creation - At any time (no schedule defined),
Automerge - At any time (no schedule defined).

🚦 **Automerge**: Disabled by config. Please merge this manually once you
are satisfied.

♻ **Rebasing**: Whenever PR becomes conflicted, or you tick the
rebase/retry checkbox.

🔕 **Ignore**: Close this PR and you won't be reminded about this update
again.

---

- [ ] <!-- rebase-check -->If you want to rebase/retry this PR, check
this box

---

This PR has been generated by [Renovate
Bot](https://togithub.com/renovatebot/renovate).

<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy40MjUuMSIsInVwZGF0ZWRJblZlciI6IjM3LjQyNS4xIiwidGFyZ2V0QnJhbmNoIjoibWFpbiIsImxhYmVscyI6WyJUZWFtOkNsb3VkIFNlY3VyaXR5IiwiYmFja3BvcnQ6YWxsLW9wZW4iLCJyZWxlYXNlX25vdGU6c2tpcCJdfQ==-->

Co-authored-by: elastic-renovate-prod[bot] <174716857+elastic-renovate-prod[bot]@users.noreply.github.com>

jennypavlova committed:
[ECO][Inventory v2] Hosts entity summary endpoint changes (#203617)

## Summary

Closes #202300

This PR changes the entity client function to v2 (`searchEntities`) in
`getLatestEntity`. After the change to use `v2.searchEntities` the
parameters are also updated to include the time range (`start` and `end`
are required)

## Testing
~- We can create some definitions manually- in the Kibana DEV tools: ~ -
Not needed after we merged the V2 PR

- In a local environment enable the entities feature flag ( it should be
a clean env as the entities should not be enabled before ):
<img width="1911" alt="image"
src="https://github.com/user-attachments/assets/75d6f77d-5039-41ca-80ca-34c3bf99844e"
/>

- Some hosts and containers are required - oblt cluster/metricbeat or
   - Create hosts using synthtrace: 
       ```
node scripts/synthtrace infra_hosts_with_apm_hosts
--scenarioOpts.numInstances=20
       ```
   - Create containers using synthtrace: 
       ```
       node scripts/synthtrace infra_docker_containers.ts
       ```

- In the UI 
- Open asset details view for hosts and containers and check the summary
endpoint response:
  ⚠️ Updated: 


![image](https://github.com/user-attachments/assets/27683b74-f0b5-43a0-9a8f-98cd2a61e68e)

- If the entities FF is disabled (default: no `logs` should be part of
the `sourceDataStreams`):


![image](https://github.com/user-attachments/assets/7b8851b4-514c-4fc7-ab84-720b2ccb16ae)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: Sergi Romeu <sergi.romeu@elastic.co>

elena-shostak committed:
Fleet Privileges Display (#204402)

## Summary

Fixed privileges display for features/subFeatures that require all
spaces.

### Before
Role privileges display for only `Default` space selected

<img width="728" alt="Screenshot 2024-12-17 at 13 32 17"
src="https://github.com/user-attachments/assets/151b7012-aa1a-430c-be22-cc91e64362e3"
/>

Privileges summary display for only `Default` space selected

<img width="471" alt="Screenshot 2024-12-17 at 13 32 50"
src="https://github.com/user-attachments/assets/964c2223-163d-4081-a37d-196f5df5df5c"
/>

### After
Role privileges display for only `Default` space selected

<img width="739" alt="Screenshot 2024-12-17 at 13 30 00"
src="https://github.com/user-attachments/assets/0f98a9d7-211d-46ec-82c6-25d29a44be6b"
/>

Privileges summary display for only `Default` space selected

<img width="569" alt="Screenshot 2024-12-17 at 13 30 19"
src="https://github.com/user-attachments/assets/932771fd-6486-4b7e-9de5-6cd34ab74dc9"
/>

### How to test
With `Default` space:
1. Navigate to Creating a new Role and assign Kibana privileges.
2. Set the Spaces to `Default` Space and the privilege level to All.
3. Navigate to Management category and verify that Fleet is set to
`None`.
4. Click on "View privilege summary" and verify that Fleet is set to
`None`.

With `*All Spaces`:
1. Navigate to Creating a new Role and assign Kibana privileges.
2. Set the Spaces to `*All Spaces` and the privilege level to All.
3. Navigate to Management category and verify that Fleet is set to `All`
4. Click on "View privilege summary" and verify that Fleet is set to
`All`


### Checklist

Check the PR satisfies following conditions. 

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

__Fixes: https://github.com/elastic/kibana/issues/194686__

## Release Note
Fixed privileges display for features/subFeatures that require all
spaces.

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>

nickpeihl committed:
[Dashboard] Add functional test for panel listing types (#205808)

Fixes #192663

## Summary

Add functional tests that verify the number of panel types registered in
the dashboards Add panel flyout. Any changes to the number of registered
panel types will trigger a review from the
`@elastic/kibana-presentation` team.

This PR also adds an additional test for panel types registered with an
advanced license.

georgianaonoleata1904 committed:
[ResponseOps][Rules]Hide number of conditions on Rule page when Rule type do not allow multiple conditions (#204385)

Closes https://github.com/elastic/kibana/issues/194809

## Summary

The "Conditions" section on the Rule page definition list is now hidden
for the rule types that don't allow multiple conditions


https://github.com/user-attachments/assets/eed1518a-91c4-453c-8294-643f86f05e30


todo: add/change tests

rmyz committed:
[APM] Fix Deep Links Cypress Flaky Test (#206000)

## Summary

Closes https://github.com/elastic/kibana/issues/205936

This PR aims to solve the flakiness of deep_links.cy test.

- Instead of doing everything under 2 tests, we run a test for each
assertion and clean everything up.
- Ensures scroll div has content to be able to scroll
- scrolls to center or bottom depending on the position

TattdCodeMonkey committed:
[Search] Remove Enterprise Search UI Apps (#205634)

## Summary

Removing app search & workplace search kibana applications from
`enterprise_search` plugin. This will be the first of many PRs to remove
code related to the enterprise search node.

### Checklist

- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: Elastic Machine <elasticmachine@users.noreply.github.com>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

viduni94 committed:
[Obs AI Assistant] Unskip user instructions test (#205656) (#205897)

Closes https://github.com/elastic/kibana/issues/205656

## Summary

### Problem
There seems to be a race condition and state contamination between tests
in the KB user instructions tests.

### Solution
- Clear knowledge base and conversations after tests
- Remove title interceptor --> we don't need title interception for
these tests

### Checklist

- [x] [Flaky Test
Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was
used on any tests changed
- [x] The PR description includes the appropriate Release Notes section,
and the correct `release_note:*` label is applied per the
[guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

jedrazb committed:
[Search serverless / Connectors FTRs] Update test role to developer (#206067)

## Summary

`viewer` role is not sufficient to call Connector APIs with new change
https://github.com/elastic/elasticsearch/pull/119863

Update the FTR tests to use developer role for testing

wajihaparvez committed:
[Docs] Add wolfi image info (#205022)

## Summary

Adds information about using Wolfi container images to the Install with
Docker page.

Closes: [#594](https://github.com/elastic/platform-docs-team/issues/594)

js-jankisalvi committed:
[ResponseOps][Rules] Allow to set notify when and throttle at rule level (#203235)

## Summary

Fixes https://github.com/elastic/kibana/issues/199494

This PR allows to create and update rule with `notify_when` and
`throttle` attributes at rule level instead of `action.frequency` level.

### How to test
- create a rule via api where `notify_when` and `throttle` attributes
are at rule level
  <details><summary>Example</summary>
  Note: use your existing connector for action
  
  ```typescript
  POST kbn:/api/alerting/rule
  {
    "tags": [],
    "params": {
      "searchConfiguration": {
        "query": {
          "query": "",
          "language": "kuery"
        },
        "index": "ff959d40-b880-11e8-a6d9-e546fe2bba5f"
      },
      "timeField": "order_date",
      "searchType": "searchSource",
      "timeWindowSize": 5,
      "timeWindowUnit": "d",
      "threshold": [
        10
      ],
      "thresholdComparator": ">",
      "size": 100,
      "aggType": "count",
      "groupBy": "all",
      "termSize": 5,
      "excludeHitsFromPreviousRun": false,
      "sourceFields": []
    },
    "schedule": {
      "interval": "1m"
    },
    "consumer": "stackAlerts",
    "name": "ES query rule from devtools",
    "rule_type_id": ".es-query",
    "notify_when": "onThrottleInterval",
    "throttle": "1h",
    "actions": [
      {
        "group": "query matched",
        "id": "ad923c8a-d27d-41a9-8c71-d33d94db4abb",
        "params": {
          "documents": [
            {
              "name": "{{rule.name}}"
            }
          ]
        }
      }
    ],
    "alert_delay": {
      "active": 1
    }
  }
  ``` 
  </details> 

- open the created rule
- go to `settings` tab 
- verify that dropdowns reflect notify_when and throttle value correctly
- update the rule via UI and verify the same
- update the rule via API and verify the same

### Flaky test runner: 
https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/7684

### Checklist

Check the PR satisfies following conditions. 

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Zacqary committed:
Fix stylelint

Zacqary committed:
Add missing disabled actions warning

Zacqary committed:
Add missing svg typedef

darnautov committed:
[ML] Transforms: Improve messages for recovered alerts  (#205721)

ashokaditya committed:
[SecuritySolutions][Endpoint] Microsoft defender for Endpoint response actions API (#205097)

## Summary

- Adds response actions client/APIs for isolate and release actions for
Microsoft Defender for Endpoint
- The feature is behind a feature flag
`responseActionsMSDefenderEndpointEnabled`

Zacqary committed:
Add rule flyout tests

Zacqary committed:
Add show request buttons

elasticmachine commented:
Pinging @elastic/response-ops (Team:ResponseOps)

jasonrhodes committed:
Adds notification settings for unsupported ftr tests (synthetics, inventory) (#206085)

## Summary

Based on [work by @v1v to notify Slack teams when there are unsupported
FTR failures](https://github.com/elastic/kibana/pull/205260), I've added
the remaining ones for synthetics and inventory tests here.

markov00 committed:
[Lens] Align Lens style with Borealis (#204839)

## Summary

This PR updates the style of Lens to align it with the new Borealis
theme.
It covers the set of tasks in #203050 within the `Lens editor panels`
list.

Comments are applied in order as in the mentioned issue

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

legrego committed:
Dependency ownership refinements (#205937)

## Summary

This PR assigns owners for dependencies that are not yet claimed. These
assignments were a "best effort", and will likely need tweaking once
downstream teams start receiving renovate PRs.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265743)
* Commit: 2e5db10feee6e5d623dc5e4e1178b2efd36dd2c9

### Failed CI Steps
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/265743#01944c32-640f-49d5-988a-426ab99485bf)
* [Jest Tests #10](https://buildkite.com/elastic/kibana-pull-request/builds/265743#01944c57-5f18-4fe9-b5ac-ca0650257482)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265743#01944c32-640f-49d5-988a-426ab99485bf) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265743/jobs/01944c32-640f-49d5-988a-426ab99485bf/artifacts/01944c57-3f23-43eb-958a-4dc3e88c2b76) Jest Tests #<span></span>10 / useRuleFormHorizontalSteps renders correctly
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/265743#01944c57-5f18-4fe9-b5ac-ca0650257482) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/265743/jobs/01944c57-5f18-4fe9-b5ac-ca0650257482/artifacts/01944c5c-a96f-458b-836e-ebf378f8ebea) Jest Tests #<span></span>10 / useRuleFormHorizontalSteps renders correctly

### History
* :broken_heart: [Build #265425](https://buildkite.com/elastic/kibana-pull-request/builds/265425) failed 18462f5253a4856894e33e223713c6536d405c61

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265743","number":265743,"commit":"2e5db10feee6e5d623dc5e4e1178b2efd36dd2c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265425","number":265425,"commit":"18462f5253a4856894e33e223713c6536d405c61"}],"number":265743}
buildkite-pr-comment-->

Zacqary committed:
Fix Jest

Zacqary committed:
Remove waitFor from tests

Zacqary committed:
Commit suggestions

Zacqary committed:
Clean up navigation functions

Zacqary committed:
Add getStepStatus function and convert reportOnBlur to JSX component

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

Zacqary commented:
Bad rebase, closing

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/265824)
* Commit: f434d485397159d944f63186d4dad1ffb97bd2fe

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/265824#01944ce9-b0f9-45d9-a52a-7ed383dfd8f3)

### History
* :broken_heart: [Build #265743](https://buildkite.com/elastic/kibana-pull-request/builds/265743) failed 2e5db10feee6e5d623dc5e4e1178b2efd36dd2c9
* :broken_heart: [Build #265425](https://buildkite.com/elastic/kibana-pull-request/builds/265425) failed 18462f5253a4856894e33e223713c6536d405c61

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265824","number":265824,"commit":"f434d485397159d944f63186d4dad1ffb97bd2fe"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265743","number":265743,"commit":"2e5db10feee6e5d623dc5e4e1178b2efd36dd2c9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/265425","number":265425,"commit":"18462f5253a4856894e33e223713c6536d405c61"}],"number":265824}
buildkite-pr-comment-->
